### PR TITLE
Add man page

### PR DESCRIPTION
**Pull request by tastytea**
This adds a man page which lists command line options, key bindings and
commands. A new option `MAN` is added, enabled by default. The man page
is only generated on UNIX-like systems (according to CMake).¹

The man pages are generated with either `a2x` (from asciidoc) or
`asciidoctor`. If both are not found, a fatal error is signaled.

¹ This apparently includes Apple OSs. I'm not sure if man pages are a thing on Apple?

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -5,6 +5,7 @@ option(CI_BUILD "Set when building in CI. Enables -Werror where possible" OFF)
 option(ASAN "Compile with address sanitizers" OFF)
 option(QML_DEBUGGING "Enable qml debugging" OFF)
 option(COMPILE_QML "Compile Qml. It will make Nheko faster, but you will need to recompile it, when you update Qt." OFF)
+option(MAN "Build man page" ON)
```
Comment (deepbluev7) [inline CMakeLists.txt:8]: I would just set this similarly to how we do it for the VOIP flag:

```
set(VOIP_DEFAULT ON)
if (APPLE OR WIN32)
	set(VOIP_DEFAULT OFF)
endif()
```

Maybe `UNIX AND NOT APPLE`

Users can still override it, if they want in that case. But .dmg's on macOS usually ship no man page, neither do flatpaks, afaik. So it makes sense to disable it by default on macOS at least.

File: man/nheko.1.adoc.in [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+same time and start multiple instances of nheko.
+
+== Keyboard shortcuts
+
+=== Room list
+
+*Ctrl-Up/Ctrl-Down*::
+Navigate within the room list.
+
+*Ctrl+k*::
```
Comment (deepbluev7) [inline man/nheko.1.adoc.in:47]: Why is this lowercase?

File: man/nheko.1.adoc.in [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+Converts a direct conversation to a normal room.
+
+== FILES
+
+*Configuration file*::
+`${XDG_CONFIG_HOME:-~/.config}/nheko/nheko.conf`
+
+*Log file*::
+`${XDG_CACHE_HOME:-~/.cache}/nheko/nheko/nheko.log`
+
```
Comment (deepbluev7) [inline man/nheko.1.adoc.in:185]: You might want to add the database location as well as the media cache? Or is it intentional to only list config and log file?

File: CMakeLists.txt (hunk 1 above)
Comment (deepbluev7) [reply CMakeLists.txt:8]: Actually, a few parts of the manpage look to be UNIX AND NOT APPLE specific to me, like the file locations.

File: man/nheko.1.adoc.in (hunk 2 above)
Comment (tastytea) [reply man/nheko.1.adoc.in:47]: Copied from <https://github.com/Nheko-Reborn/nheko/wiki/Shortcuts>. 😊 I can make them all uppercase?

**Comment by deepbluev7** [comment]
Do you want to fix the CI or should I do that?

File: man/nheko.1.adoc.in (hunk 3 above)
Comment (tastytea) [reply man/nheko.1.adoc.in:185]: No, I just forgot to add the other files. :sweat_smile:

File: man/nheko.1.adoc.in (hunk 2 above)
Comment (deepbluev7) [reply man/nheko.1.adoc.in:47]: Well, if it is not too much effort, that would be nice, I think

**Comment by tastytea** [comment]
> Do you want to fix the CI or should I do that?

I can do that. It's only `.gitlab-ci.yml`, right?

**Comment by deepbluev7** [comment]
> > Do you want to fix the CI or should I do that?
> 
> I can do that. It's only `.gitlab-ci.yml`, right?

Yes, Windows shouldn't fail... I hope :3

**Comment by tastytea** [comment]
Messed up the `UNIX AND NOT APPLE` condition, should work now.

**Comment by tastytea** [comment]
Hmm, the flatpak-recipe can't find `a2x`, but [it is installed](https://packages.ubuntu.com/jammy/all/asciidoc-base/filelist). I think It may have something to do with `flatpak-builder`? Do I need to add something to `io.github.NhekoReborn.Nheko.yaml`?

**Review by deepbluev7 — CHANGES_REQUESTED** [review]

**Comment by deepbluev7** [comment]
I forgot to submit my comment again. Just disable the man for flatpaks, it is very cumbersome to access anyway.

File: man/nheko.1.adoc.in [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+*Database*::
+`${XDG_DATA_HOME:-~/.local/share}/nheko/nheko/*/data.mdb`
+
+*media cache*::
```
Comment (deepbluev7) [inline man/nheko.1.adoc.in:189]:
```suggestion
*Media cache*::
```

File: man/nheko.1.adoc.in [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+
+=== Room list
+
+*Ctrl-Up/Ctrl-Down*::
+Navigate within the room list.
+
+*Ctrl+K*::
+Search and select rooms from the room list.
+
+*Alt+A*, *Ctrl+Shift+A*::
```
Comment (deepbluev7) [inline man/nheko.1.adoc.in:50]:
```suggestion
*Ctrl-K*::
Search and select rooms from the room list.

*Alt-A*, *Ctrl-Shift-A*::
```

File: man/nheko.1.adoc.in [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+*Alt-Up/Alt-Down*::
+Reply to a message.
+
+*Up/Down*::
+Edit a message.
+
+*Ctrl-E*::
+Edit message currently selected as reply.
+
+*Alt+F*::
```
Comment (deepbluev7) [inline man/nheko.1.adoc.in:67]:
```suggestion
*Alt-F*::
```

**Review by deepbluev7 — APPROVED** [review]

**Comment by deepbluev7** [comment]
Thank you! <3

**Pull request merged**